src/psyclone/transformations.py

```diff
@@ -356,7 +355,110 @@ def apply(self, node, options=None):
             self.omp_nogroup = current_nogroup
 
 
-class OMPDeclareTargetTrans(Transformation):
+class MarkRoutineForGPUMixin:
+
+    def validate_it_can_run_on_gpu(self, node, options):
+        '''
+        Check that an this node can be marked as available to be called on
+        GPU.
+        '''
+        force = options.get("force", False) if options else False
+
+        if not isinstance(node, Kern) and not isinstance(node, Routine):
```

Comment:
`not isinstance(node, (Kern, Routine)):`

Reply:
Done